internal/app/concurrent_proxy_stage_test.go

```diff
@@ -208,6 +237,32 @@ func (s *ConcurrentProxyStage) all_the_user_responses_should_have_the_right_stat
 	return s
 }
 
+func (s *ConcurrentProxyStage) the_second_user_response_should_have_the_right_status_code() *ConcurrentProxyStage {
+	c := 0
+	for _, res := range s.userResponses {
+		if res.StatusCode == http.StatusConflict {
+			c++
+		}
+	}
+	s.assert.Equal(1, c)
+	return s
+}
+
+func (s *ConcurrentProxyStage) the_second_user_response_should_have_a_modified_body() *ConcurrentProxyStage {
+	c := 0
+	for _, res := range s.userResponses {
```

Comment:
could we try to make tests a bit more reliable, for example:
- verify that there is exactly 2 responses beforehand
- verify body and status code of both responses, one of them being modified the other one not

Reply:
done